staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -155,18 +161,28 @@ func (r *proxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	proxyRoundTripper = transport.NewAuthProxyRoundTripper(user.GetName(), user.GetGroups(), user.GetExtra(), proxyRoundTripper)
 
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.APIServerTracing) && !upgrade {
+		tracingWrapper := tracing.WrapperFor(r.tracerProvider)
+		proxyRoundTripper = tracingWrapper(proxyRoundTripper)
+	}
+
 	// If we are upgrading, then the upgrade path tries to use this request with the TLS config we provide, but it does
 	// NOT use the proxyRoundTripper.  It's a direct dial that bypasses the proxyRoundTripper.  This means that we have to
 	// attach the "correct" user headers to the request ahead of time.
 	if upgrade {
 		transport.SetAuthProxyHeaders(newReq, user.GetName(), user.GetGroups(), user.GetExtra())
 	}
 
-	handler := proxy.NewUpgradeAwareHandler(location, proxyRoundTripper, true, upgrade, &responder{w: w})
+	upgradeHandler := proxy.NewUpgradeAwareHandler(location, proxyRoundTripper, true, upgrade, &responder{w: w})
 	if r.rejectForwardingRedirects {
-		handler.RejectForwardingRedirects = true
+		upgradeHandler.RejectForwardingRedirects = true
 	}
 	utilflowcontrol.RequestDelegated(req.Context())
+
+	var handler http.Handler = upgradeHandler
```

Comment:
the usual pattern is:

```
handler := ....
handler := WithSomething(handler)
handler := WithSomethingElse(handler)

handler.ServeHTTP(...)
```

Reply:
I looked through the codebase on how others handled this, it seems like some alternatives could be 

`handler = http.Handler(upgradeHandler)` like in https://github.com/kubernetes/kubernetes/blob/d9cd8eed52211c5ce64807dbf9a6355b8b2a8bc5/staging/src/k8s.io/kubectl/pkg/proxy/proxy_server.go#L214-L220 or creating a new wrapper function that returns a `http.Handler` interface. `proxy.NewUpgradeAwareHandler` is used in too many places to directly change the return type there.

I'm okay with how it is currently though, will defer to @sttts

Reply:
the related code was removed, so marking this as resolved